src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -162,18 +207,39 @@ DlgPrefWaveform::~DlgPrefWaveform() {
 void DlgPrefWaveform::slotUpdate() {
     WaveformWidgetFactory* factory = WaveformWidgetFactory::instance();
 
+    bool isAccelerationEnabled = false;
     if (factory->isOpenGlAvailable() || factory->isOpenGlesAvailable()) {
         openGlStatusData->setText(factory->getOpenGLVersion());
+        useAccelerationCheckBox->setEnabled(true);
+        isAccelerationEnabled = m_pConfig->getValue(
+                                        ConfigKey("[Waveform]", "use_hardware_acceleration"),
+                                        factory->preferredBackend()) !=
+                WaveformWidgetBackend::None;
```

Comment:
I think you are misinterpreting the code slightly :) 
Here a comment about `use_hardware_acceleration`
```
    // On the UI, hardware acceleration is a boolean (0 => software rendering, 1
    // => hardware acceleration), but in the setting, we keep the granularity so
    // in case of issue when we release, we can communicate workaround on
    // editing the INI file to target a specific rendering backend. If no
    // complains come back, we can convert this safely to a backend eventually.
```
This means that `m_pConfig->getValue()` will return an integer, which is expected to be a `WaveformWidgetBackend`. The idea I had in mind when I did this was to still keep the field core behaviour as a toggle/boolean, but also allow anyone to change the INI to actually force another rendering backend, not available via the UI (which only all Allshader - accelerated - or software/Qt - fallback). This is here as a safety measure so if someone report an issue during the beta we can easily compare behaviour and looks but also instruct them with a workaround till the fix becomes available.

Reply:
Yeah I only saw that comment later and forgot to delete this one again. Sorry for the noise :)